src/os/file_other.go

```diff
@@ -22,6 +22,19 @@ const isOS = false
 // number. It implements the FileHandle interface.
 type stdioFileHandle uint8
 
+// file is the real representation of *File.
+// The extra level of indirection ensures that no clients of os
+// can overwrite this data, which could cause the finalizer
+// to close the wrong file descriptor.
+type file struct {
+	handle FileHandle
+	name   string
+}
+
+func NewFile(fd FileHandle, name string) *File {
```

Comment:
This differs from Go in taking an interface for `fd` instead of a `uintptr`, which seems to break if passing, e.g., file descriptor 3 as done in [Go 1.18's fuzzing code](https://github.com/golang/go/blob/d588f487703e773ba4a2f0a04f2d4141610bff6b/src/internal/fuzz/sys_posix.go#L61) (cf. #2515)

Reply:
IIRC that was taken straight from upstream?  Maybe upstream has changed, or maybe I'm misremembering.

Either way, please file a bug and we can sort it.

Reply:
Well, it's been that way for 10 years, it appears. I'll just open a PR as I've already made the changes locally.